### PR TITLE
Use relative $refs in external property file schema

### PR DESCRIPTION
**Pull request by lgolding**

### CODE REVIEW

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

File: src/Sarif/Sarif.csproj [hunk 1]
```diff
@@ -42,15 +42,16 @@
     <!--
     Marking these items as content causes them to be included in the NuGet package.
     -->
-    <Content Include="Schemata\sarif-schema.json" />
+    <Content Include="Schemata\sarif-schema-2.1.0-rtm.0.json" />
```
Comment (lgolding) [inline src/Sarif/Sarif.csproj]: >sarif-schema-2.1.0-rtm.0.json [](start = 31, length = 29)

Note how many different places this appears. We must either DRY it out or go through this again if we need rtm.1.

File: src/Test.FunctionalTests.Sarif/SarifValidatorTests.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public class SarifValidatorTests
     {
-        public const string JsonSchemaFile = "sarif-schema.json";
+        public const string JsonSchemaFile = "sarif-schema-2.1.0-rtm.0.json";
```
Comment (harleenkohli) [inline src/Test.FunctionalTests.Sarif/SarifValidatorTests.cs]: >2.1.0-rtm.0 [](start = 59, length = 11)

should we use VersionConstants.SchemaVersionAsPublishedToSchemaStoreOrg here? (and other files which use this literal) this way updating to newer release will be a few steps less.

File: src/Sarif/Autogenerated/ToolComponent.cs [hunk 3]
```diff
@@ -101,7 +101,7 @@ public SarifNodeKind SarifNodeKind
         public string DottedQuadFileVersion { get; set; }
 
         /// <summary>
-        /// A string specifying the UTC date (and optionally, the time) of the component�s release.
```
Comment (harleenkohli) [inline src/Sarif/Autogenerated/ToolComponent.cs]: >� [](start = 88, length = 1)

yuck! thanks for fixing this (and finding thisss!)

File: src/Test.FunctionalTests.Sarif/SarifValidatorTests.cs (hunk 2 above)
Comment (lgolding) [reply src/Test.FunctionalTests.Sarif/SarifValidatorTests.cs]: Good idea!

---
In reply to: [285310264](https://github.com/microsoft/sarif-sdk/pull/1493#discussion_r285310264) [](ancestors = 285310264)

File: src/Sarif.Multitool/ValidateCommand.cs [hunk 4]
```diff
@@ -46,7 +46,7 @@ protected override SarifValidationContext CreateContext(ValidateOptions options,
         }
 
         private static string s_CurrentSchemaLocation = 
-            Path.Combine(Path.GetDirectoryName(typeof(ValidateCommand).Assembly.Location), "sarif-schema.json");
+            Path.Combine(Path.GetDirectoryName(typeof(ValidateCommand).Assembly.Location), "sarif-schema-2.1.0-rtm.0.json");
```
Comment (lgolding) [inline src/Sarif.Multitool/ValidateCommand.cs]: >- [](start = 104, length = 1)

Here as well `VersionConstants.SchemaVersionAsPublishedToSchemaStoreOrg`.

Comment (lgolding) [reply src/Sarif.Multitool/ValidateCommand.cs]: And in the project files I can just use the property from build.props! Hold on for another iteration.

---
In reply to: [285310953](https://github.com/microsoft/sarif-sdk/pull/1493#discussion_r285310953) [](ancestors = 285310953)

File: src/Sarif/Sarif.csproj [hunk 5]
```diff
@@ -42,15 +42,16 @@
...
+    <Content Include="Schemata\sarif-external-property-file-schema-$(SchemaVersionAsPublishedToSchemaStoreOrg).json" />
 
     <!-- Link=false: Don't display this item in Solution Explorer. -->
     <!-- This item will be added to the NuGet package, however. -->
     <Content Include="..\ReleaseHistory.md" Link="false" />
   </ItemGroup>
 
   <ItemGroup>
-    <JsonSchemaFile Include="Schemata\sarif-schema.json">
+    <JsonSchemaFile Include="Schemata\sarif-schema-2.1.0-rtm.0.json">
```
Comment (lgolding) [inline src/Sarif/Sarif.csproj]: >2.1.0-rtm.0 [](start = 51, length = 11)

Darn, missed a place

**Pull request merged**